src/main/java/io/antmedia/streamsource/StreamFetcher.java

```diff
@@ -202,9 +203,20 @@ public void run() {
 
 
 				if (result.isSuccess()) {
+					boolean audioOnly = false;
+					if(inputFormatContext.nb_streams() == 1) {
+						audioOnly  = (inputFormatContext.streams(0).codecpar().codec_type() == AVMEDIA_TYPE_AUDIO);
+					}
 					
-					muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope);
+					if(audioOnly) {
+						muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope, false);
+						muxAdaptor.setAudioOnly(audioOnly);
+					}
+					else {
+						muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope);
+					}
```

Comment:
Before taking care of below comment, please take a look at the comment in here
https://github.com/ant-media/Ant-Media-Server-Common/commit/10b6a25ae7b292d051f788feffe848569dd5584d


It will be better if we have less if else statement. I mean below code
```
if(audioOnly) {
muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope, false)
muxAdaptor.setAudioOnly(audioOnly);
}
else {
muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope);
}
```
can be simplified to 
````
muxAdaptor = MuxAdaptor.initializeMuxAdaptor(null,true, scope, !audioOnly); 
muxAdaptor.setAudioOnly(audioOnly);
````